--- a/source/postcard/src/max_size.rs
+++ b/source/postcard/src/max_size.rs
@@ -333,7 +333,7 @@ const fn varint_size(max_n: usize) -> usize {
     // We add (BITS_PER_BYTE - 1), to ensure any integer divisions
     // with a remainder will always add exactly one full byte, but
     // an evenly divided number of bits will be the same
-    let roundup_bits = bits + (BITS_PER_VARINT_BYTE - 1);
+    let roundup_bits = bits + (BITS_PER_BYTE - 1);

     // Apply division, using normal "round down" integer division
     roundup_bits / BITS_PER_VARINT_BYTE
