--- a/source/postcard/src/de/deserializer.rs
+++ b/source/postcard/src/de/deserializer.rs
@@ -477,12 +477,12 @@ impl<'de, 'a, F: Flavor<'de>> de::Deserializer<'de> for &'a mut Deserializer<'de
     fn deserialize_tuple_struct<V>(
         self,
         _name: &'static str,
-        len: usize,
+        _len: usize,
         visitor: V,
     ) -> Result<V::Value>
     where
         V: Visitor<'de>,
     {
-        self.deserialize_tuple(len, visitor)
+        self.deserialize_seq(visitor)
     }

     #[inline]
